src/core/packet.lua

```diff
@@ -109,3 +114,21 @@ function preallocate_step()
    packet_allocation_step = 2 * packet_allocation_step
 end
 
+function dump(p, w)
```

Comment:
Agree, the minimum requirement is the `physical` function. Everything else can go away.